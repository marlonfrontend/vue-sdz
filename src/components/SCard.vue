<template>

</template>

<script>
export default {
  name: 'SCard'
}
</script>

<style lang="scss">

</style>
