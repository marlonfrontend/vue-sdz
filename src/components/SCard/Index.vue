<template>
  <div class="s-card">
    card
  </div>
</template>

<script>
export default {
  name: 'SCard'
}
</script>

<style lang="scss">
.s-card { background: red; }
</style>
