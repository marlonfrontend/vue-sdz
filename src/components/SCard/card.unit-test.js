import { mount, shallowMount } from '@vue/test-utils'

import SCard from './Index.vue'

/* eslint-disable */
describe('SCard', () => {
  let cmp;

  beforeEach(() => {
    cmp = mount(SCard, {
      propsData: {
        backgroundColor: 'red',
        borderColor: 'blue'
      },
      slots: {
        default: `Texto do card`
      }
    });
  });


  test('is instantiated', () => {
    expect(cmp.exists()).toBeTruthy()
  })
  

  test("has the expected html structure", () => {
    expect(cmp.element).toMatchSnapshot();
  })

  test('check card text', () => {

    const span = cmp.find('div')
    expect(span.exists()).toBe(true)
    expect(span.text()).toBe('Texto do card')

  })


  test('check background color', () => {

    expect(cmp.attributes().style).toContain('red')

  })

  test('check border color', () => {

    expect(cmp.attributes().style).toContain('blue')

  })

  test('check class', () => {

    expect(cmp.classes('s-card')).toBe(true)
    expect(cmp.classes()).toContain('s-card')

  })


})
