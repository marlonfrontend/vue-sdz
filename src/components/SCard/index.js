import SCard from './Index.vue'

export default Vue => {
  Vue.component(SCard.name, SCard)
}
