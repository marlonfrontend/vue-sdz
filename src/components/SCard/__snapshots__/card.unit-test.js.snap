// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SCard has the expected html structure 1`] = `
<div
  class="s-card"
  style="background: red; border-left: 4px solid blue;"
>
  Texto do card
</div>
`;
